### PR TITLE
Add dataset.py for Prompt ASR

### PR DESCRIPTION
**Pull request by marcoyang1998**
This PR adds a dataset.py for Prompt ASR.

As the initial step, support randomly selecting one text as transcription from a list of texts while constructing batches.

### CODE REVIEW

File: egs/libri_heavy/ASR/local/dataset.py [hunk 1]
```diff
@@ -24,7 +24,7 @@ def __init__(
         cut_transforms: List[Callable[[CutSet], CutSet]] = None,
         input_transforms: List[Callable[[torch.Tensor], torch.Tensor]] = None,
         input_strategy: BatchIO = PrecomputedFeatures(),
-        text_sampling_func: Callable = [str],
+        text_sampling_func: Callable[[List[str]], str] = None,
```
Comment (csukuangfj) [inline egs/libri_heavy/ASR/local/dataset.py:27]:
```suggestion
        text_sampling_func: Optional[Callable[[List[str]], str]] = None,
```

File: egs/libri_heavy/ASR/local/dataset.py [hunk 2]
```diff
@@ -104,7 +104,9 @@ def __getitem__(self, cuts: CutSet) -> Dict[str, Union[torch.Tensor, List[str]]]
             "supervisions": default_collate(
                 [
                     {
-                        "text": self.text_sampling_func(supervision.texts),
+                        "text": supervision.texts
```
Comment (csukuangfj) [inline egs/libri_heavy/ASR/local/dataset.py:107]:
```suggestion
                        "text": supervision.text
```

**Comment by pingfengluo** [comment]
@marcoyang1998 hi ，can you merge this PR？

**Comment by marcoyang1998** [comment]
Hi @pingfengluo, we will probably open another PR for PromptASR.

**Comment by minmin0** [comment]
hello, @marcoyang1998 @danpovey , is there any update for this speech-prompt？Is it available for use？

**Comment by marcoyang1998** [comment]
Hi @minmin0, we are going to update the PR this week.

**Comment by minmin0** [comment]
> Hi @minmin0, we are going to update the PR this week.

@marcoyang1998 cool

**Comment by minmin0** [comment]
> update

hi @marcoyang1998 , is there any update for the speech-prompt PR？

**Comment by minmin0** [comment]
> 



> Hi @minmin0, we are going to update the PR this week.

hi @marcoyang1998 is there any update for the speech-prompt PR？

**Comment by marcoyang1998** [comment]
@minmin0 

Please see this branch: https://github.com/marcoyang1998/icefall/tree/libriheavy_prompt_asr

**Comment by marcoyang1998** [comment]
Closing this as a new PR has been made #1250

**Pull request closed without merging**